### ISSUE TITLE
Hand-crank charger not recharging and leading to a bug.

### ISSUE THREAD

**Issue by Proluiz** [CleverRaven/Cataclysm-DDA#41644]
# Describe the bug
Seen related to #41085 || Not a crash.
Hand-crank charger loaded with am empty light battery wont recharge it and attemping to do it lead to a bug.

# Steps To Reproduce
1. Get an Hand-crank charger and a empty light battery.
2. Reload the hand-crank charger with the light battery.
3. Attemp to Activate the Hand-crank to recharge it.

# Expected behavior
Expected to recharge the light battery, but intead the debug screen happens and if you told to ignore future errors the game will attemp to reload the battery whitout sucess, until you manually cancel the process.

# Versions and configuration
 - OS: Windows 7
 - Game Version: 0.E-3553-ge888413
 - Mods loaded: [Dark Days Ahead, StatsThroughSkills, StatsThroughKills, DisableNPCNeeds]

# Additional context
Here's the debug.log file.
[debug.log](https://github.com/CleverRaven/Cataclysm-DDA/files/4841528/debug.log)

**Comment by woutershep** [comment]
The debug message (so that people can find it by searching) is:

ERROR : src/item.cpp:632 [item& item::ammo_set(const itype_id&, int)] Tried to set invalid ammo of battery for hand-crank charger

**Comment by wapcaplet** [comment]
Confirmed, following the steps - upon `a`ctivating the hand-crank charger with depleted battery in it, I get this error:

```
 DEBUG    : Tried to set invalid ammo of batteries for hand-crank charger
 FUNCTION : item &item::ammo_set(const itype_id &, int)
 FILE     : src/item.cpp
 LINE     : 627
```

- OS: Linux
    - OS Version: Distributor ID: Pop; Description: Pop!_OS 20.04 LTS; Release: 20.04; Codename: focal; 
- Game Version: 0.E-3531-gff1ff7ea9f [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

**Comment by Duck13Tales** [comment]
Just bumped onto this bug.

I am using using the CDDA 0.E-b10798 with the mods:

DDA
Disable NPC needs
Stats through skills
Stats through kills
SpeedyDex

**Comment by kevingranade** [comment]
Fixed by #41917
